packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -655,6 +657,10 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
         this.autoscaledPoolOptions = { ...autoscaledPoolOptions, ...basicCrawlerAutoscaledPoolConfiguration };
     }
 
+    protected isProxyError(error: Error) {
+        return ROTATE_PROXY_ERRORS.some((x: string) => (this._getMessageFromError(error) as any)?.includes(x));
+    }
+
     protected isRequestBlocked(_crawlingContext: Context) {
         throw new Error('the "isRequestBlocked" method is not implemented in this crawler.');
     }
```

Comment:
Should we conflate proxy errors and request blocking an process them using the same mechanism (`SessionError`, separate retry count?), with request blocking still being opt-in?

Reply:
makes sense